sky/packages/sky/lib/src/rendering/box.dart

```diff
@@ -91,7 +107,8 @@ class BoxConstraints extends Constraints {
     );
   }
 
-  BoxConstraints apply(BoxConstraints constraints) {
+  /// Returns a new box constraints whose values respect the given constraints
```

Comment:
they do more than just respect the given constraints. I mean, a copy constructor would satisfy that definition. :-) This method merges the two sets of BoxConstraints by limiting this object by the requirements described by the other, or something.